Data/WorkerInfo.py

```diff
@@ -1,10 +1,17 @@
 class WorkerInfo:
     
-    def __init__(self, hostName, portNumber, maxThreadCount, isGPUEnabled):
+    def __init__(self, hostName, portNumber, maxThreadCount, isGPUEnabled, hardwareGeneration):
         self.hostName = hostName
         self.portNumber = portNumber
         self.maxThreadCount = maxThreadCount
         self.isGPUEnabled = isGPUEnabled
+        self.hardwareGeneration = hardwareGeneration
 
     def __str__(self):
-        return "[HostName: {} || PortNumber: {} || MaxThreadCount: {} || IsGPUEnabled: {}]".format(self.hostName, self.portNumber, self.maxThreadCount, self.isGPUEnabled)
+        return "[HostName: {} || PortNumber: {} || MaxThreadCount: {} || IsGPUEnabled: {} || Hardware Generation: {}]".format(self.hostName, self.portNumber, self.maxThreadCount, self.isGPUEnabled, self.hardwareGeneration)
+
+    def __lt__(self, other):
```

Comment:
what is the purpose of this?

Reply:
The lt/gt overwrite is used when in priority queue's first element is same, the internal python implementation uses the second property (in this case workerInfo) to maintain maxHeap/minHeap property